openlibrary/plugins/upstream/addtag.py

```diff
@@ -0,0 +1,164 @@
+"""Handlers for adding and editing tags."""
+
+import web
+import json
+
+from typing import NoReturn
+
+from infogami.core.db import ValidationException
+from infogami.infobase import common
+from infogami.utils.view import add_flash_message, public
+from infogami.infobase.client import ClientException
+from infogami.utils import delegate
+
+from openlibrary.plugins.openlibrary.processors import urlsafe
+from openlibrary.i18n import gettext as _
+import logging
+
+from openlibrary.plugins.upstream import spamcheck, utils
+from openlibrary.plugins.upstream.models import Tag
+from openlibrary.plugins.upstream.addbook import get_recaptcha, safe_seeother, trim_doc
+from openlibrary.plugins.upstream.utils import render_template
+
+logger = logging.getLogger("openlibrary.tag")
+
+
+@public
+def get_tag_types():
+    return ["subject", "work", "collection"]
+
+
+class addtag(delegate.page):
+    path = '/tag/add'
+
+    def GET(self):
+        """Main user interface for adding a tag to Open Library."""
+
+        if not self.has_permission():
+            raise common.PermissionDenied(message='Permission denied to add tags')
+
+        return render_template('tag/add', recaptcha=get_recaptcha())
+
+    def has_permission(self) -> bool:
+        """
+        Can a tag be added?
+        """
+        return web.ctx.user and (
+            web.ctx.user.is_usergroup_member('/usergroup/super-librarians')
+        )
```

Comment:
```suggestion
        user = web.ctx.site.get_user()
        return user and (
            user.is_usergroup_member('/usergroup/super-librarians')
        )
```
Attempting to get the current user in this way is causing errors.  Use `get_user()` instead.